R/normalize.R

```diff
@@ -1,80 +1,97 @@
+#' validate_columns_exist
+#' 
+#' This function checks that a list of columns are present in a dataframe.
+#' 
+#' @param selected_columns A vector of strings each representing a column name
+#' @param df A dataframe to check against
+#' @return Boolean
+validate_columns_exist= function(selected_columns, df) {
+  # Check that all of selected_columns are in df
+  if(any(!selected_columns %in% colnames(df))) {
+    return(FALSE)
+  } else {
+    return(TRUE)
+  }
+}
+
 #'  normalize
 #'
 #'  takes a filtered dataframe of raw read counts and normalizes
 #'  counts using control barcodes
 #'
-#' @param X - dataframe of annotated readcounts that must include the following columns:
+#' @param X A dataframe of annotated readcounts that must include the following columns:
 #'           log2_n or n: raw readcounts or log2(n) of read counts. Computes log2_n if not present
 #'           log_dose: log2 of dose at which control barcode was spiked in, if applicable
 #'           Name: contains the name of the control barcode that the read corresponds to, or NA
 #'           profile_id: a unique identifier for each sample, as generated by filter_counts
-#' @param barcodes - a vector of control barcode Name identifiers
+#' @param id_cols a vector of columns used to identify each PCR well and to be carried forward in the pipeline.
+#'                These column names should be present in dataframe X.
+#' @param barcodes - a vector of control barcode names
 #' @param pseudocount A pseudocount to be added to the counts so that logs can be taken
-#' @return - table with additional log_normalized_n and normalized_n columns containing counts normalized to control barcodes
+#' @returns Dataframe with counts normalized to control barcodes
 #' @import tidyverse
 #' @import magrittr
-#' @return dataframe of normalized counts with the following columns
-#'         - cb_intercept: intercept of the fit
-#'         - norm_mae: median absolute error of the fit
-#'         - norm_r2: r2 value of the fit
-#'         - log2_normalized_n
-#'         - normalized_n
-normalize <- function(X, barcodes, pseudocount) {
+normalize <- function(X, id_cols, barcodes, pseudocount) {
   # Required functions
   require(tidyverse)
   require(magrittr)
   
-  # Create log2_n if it does not exist
+  # Create log2_n if it does not exist ----
```

Comment:
is there a reason we don't want to just overwrite log2_n even if it exists?

Reply:
Creat/override log2_n even if it exists